### PR TITLE
ENH: Prefer scipy.fft to scipy.fftpack in scipy.signal

### PR DESCRIPTION
**Pull request by larsoner**
Change `scipy.signal` to prefer `scipy.fft` over `scipy.fftpack` and `scipy.fft.rfft` over `np.fft.rfft`.

Have not tested locally, let's see if this works. Will push commits to fix when it inevitably doesn't.

### CODE REVIEW

File: scipy/__init__.py [hunk 1]
```diff
@@ -128,3 +121,14 @@
...
+
+    # This makes "from scipy import fft" return scipy.fft, not np.fft
+    del fft
+    from . import fft
+
+    # The existence of the scipy.fft function breaks doc builds
+    import os as _os
+    if _os.environ.get('_SCIPY_BUILDING_DOC') == 'True':
+        del fft
+        __all__.remove('fft')
```
Comment (peterbell10) [inline scipy/__init__.py]: If `scipy.fft` isn't a function anymore then this shouldn't be needed.

Comment (larsoner) [reply scipy/__init__.py]: Unfortunately I think we need to wait for #10290

Comment (peterbell10) [reply scipy/__init__.py]: I mean that if `scipy.fft` is always the module then there's no reason to do `del fft` here as it shouldn't confuse sphinx.

Comment (larsoner) [reply scipy/__init__.py]: Thanks @peterbell10 that fixed the Sphinx problem!

File: scipy/signal/signaltools.py [hunk 2]
```diff
@@ -404,15 +404,15 @@ def fftconvolve(in1, in2, mode="full", axes=None):
         in1, s1, in2, s2 = in2, s2, in1, s1
 
     # Speed up FFT by padding to optimal size for FFTPACK
-    fshape = [fftpack.helper.next_fast_len(d) for d in shape[axes]]
+    fshape = [sp_fft.next_fast_len(d) for d in shape[axes]]
     fslice = tuple([slice(sz) for sz in shape])
     # Pre-1.9 NumPy FFT routines are not threadsafe.  For older NumPys, make
```
Comment (endolith) [inline scipy/signal/signaltools.py]: Since we're not using numpy anymore the threadsafe stuff can be removed?

**Comment by endolith** [comment]
https://github.com/scipy/scipy/blob/a9dab60b053ab21da792aacace7a1b157cee9932/scipy/signal/signaltools.py#L487 should be re-tuned for the new fft implementation

**Comment by endolith** [comment]
https://github.com/scipy/scipy/blob/a9dab60b053ab21da792aacace7a1b157cee9932/scipy/fft/_helper.py#L53

also add a real or complex parameter to `next_fast_len()` and handle the higher radix sizes?

File: scipy/signal/signaltools.py (hunk 2 above)
Comment (larsoner) [reply scipy/signal/signaltools.py]: That's in #9876, I'm trying not to mix them too much

**Comment by larsoner** [comment]
> should be re-tuned for the new fft implementation

Ahh yes I'll probably need to look at the old PR to figure out how to tune it. If you have time to check and/or remember how, feel free.

> also add a real or complex parameter to next_fast_len() and handle the higher radix sizes?

I think @peterbell10 might already have an issue about this, if not feel free to open one

**Comment by peterbell10** [comment]
>I think @peterbell10 might already have an issue about this, if not feel free to open one

There was some related discussion in #10240, it may deserve it's own issue though.

**Comment by endolith** [comment]
> Ahh yes I'll probably need to look at the old PR to figure out how to tune it. If you have time to check and/or remember how, feel free.

Yeah I have the script around somewhere

**Comment by larsoner** [comment]
Turned the updating into a new issue.

Our implementation in this PR should be at least as fast as what was done before, so at worst we are performing as well as we were before (when using time-domain) and better than before (when using freq-domain), so a follow-up PR to further tune stuff should be fine, and not a blocker for this PR.

@endolith okay with you?

**Comment by endolith** [comment]
Yes that's fine

**Comment by larsoner** [comment]
@peterbell10 it might be best to merge this one first. I'll push a commit to remove the thread locking comment, then I think it should be good to merge if you're happy.

**Review by peterbell10 — APPROVED** [review]

**Comment by larsoner** [comment]
@peterbell10 do you want to click the green button? It's already a single commit so any of the three merge options are fine but I usually do "rebase and merge"

**Comment by peterbell10** [comment]
I’m getting a status 500 server error when I try to open this specific PR…

**Pull request merged**